src/batch.ts

```diff
@@ -41,35 +43,49 @@ export class Batch {
      * @param control Optional control parameters to tune the integration
      */
     constructor(run: singleBatchRun, pars: BatchPars, tStart: number, tEnd: number) {
-        this.pars = pars;
+        this.pars = { ...pars, values: [] as number[]};
```

Comment:
So you're initialising with no values in the pars, and then you'll add in the values as they are used? Could be worth updating the comment to note that